tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -742,3 +271,316 @@ def _enable_disable_volume_profiling(self):
+            "notice",
+            "alerting",
+            {'message': json.dumps(alert)}
+        )
+    )
+
+
+def sync_volumes(volumes, index, vol_options):
+    # instantiating blivet class, this will be used for
+    # getting brick_device_details
+    b = blivet.Blivet()
+
+    # reset blivet during every sync to get latest information
+    # about storage devices in the machine
+    b.reset()
+    devicetree = b.devicetree
+
+    SYNC_TTL = int(NS.config.data.get("sync_interval", 10)) + 250
+    node_context = NS.node_context.load()
+    tag_list = list(node_context.tags)
+    # Raise alerts for volume state change.
+    cluster_provisioner = "provisioner/%s" % NS.tendrl_context.integration_id
+    if cluster_provisioner in tag_list:
+        try:
+            stored_volume_status = NS._int.client.read(
+                "clusters/%s/Volumes/%s/status" % (
+                    NS.tendrl_context.integration_id,
+                    volumes['volume%s.id' % index]
+                )
+            ).value
+            current_status = volumes['volume%s.status' % index]
+            if stored_volume_status != "" and \
+                current_status != stored_volume_status:
+                msg = "Status of volume: %s " + \
+                      "changed from %s to %s" % (
+                          volumes['volume%s.name' % index],
+                          stored_volume_status,
+                          current_status
+                      )
+                instance = "volume_%s" % volumes[
+                    'volume%s.name' % index
+                ]
+                emit_event(
+                    "volume_status",
+                    current_status,
+                    msg,
+                    instance
+                )
+        except etcd.EtcdKeyNotFound:
+            pass
+
+    volume = NS.gluster.objects.Volume(
+        vol_id=volumes['volume%s.id' % index],
+        vol_type="arbiter"
+        if int(volumes['volume%s.arbiter_count' % index]) > 0
+        else volumes['volume%s.type' % index],
+        name=volumes['volume%s.name' % index],
+        transport_type=volumes['volume%s.transport_type' % index],
+        status=volumes['volume%s.status' % index],
+        brick_count=volumes['volume%s.brickcount' % index],
+        snap_count=volumes['volume%s.snap_count' % index],
+        stripe_count=volumes['volume%s.stripe_count' % index],
+        replica_count=volumes['volume%s.replica_count' % index],
+        subvol_count=volumes['volume%s.subvol_count' % index],
+        arbiter_count=volumes['volume%s.arbiter_count' % index],
+        disperse_count=volumes['volume%s.disperse_count' % index],
+        redundancy_count=volumes['volume%s.redundancy_count' % index],
+        quorum_status=volumes['volume%s.quorum_status' % index],
+        snapd_status=volumes['volume%s.snapd_svc.online_status' % index],
+        snapd_inited=volumes['volume%s.snapd_svc.inited' % index],
+    )
+    volume.save(ttl=SYNC_TTL)
```

Comment:
You mean the whole volume saving by provisioner only? how would it save the node specific bricks in that case?